### PR TITLE
[Draft] feat(web): Share or copy datetime specific URL

### PR DESCRIPTION
**Pull request by tonypls**
## Issue
We want to enable historical linking to a specific hour in the past

## Description
This PR copies the current url and selectedDatetime to the clipboard on desktop and launches the native share functionality on mobile.

I chose not to use the share api on desktop because it's not super useful, see image:
image

I think we could build our own share to LinkedIn, X etc buttons to achieve a similar experience to the mobile native share.

### CODE REVIEW

**Comment by tonypls** [comment]
@cadeban Here's a draft of the sharing functionality I've implemented so far 😃

**Comment by VIKTORVAV99** [comment]
> I think we could build our own share to LinkedIn, X etc buttons to achieve a similar experience to the mobile native share.

We do have these buttons already, but only for the ranking panel/map but it would be nice to add them elsewhere as well.

File: web/src/utils/constants.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import type { Duration } from 'date-fns';
 import { ElectricityModeType } from 'types';
 
-export const metaTitleSuffix = ' | App | Electricity Maps';
+export const metaTitleSuffix = ' - Map - Electricity Maps';
```
[the remark this answers is not in the document]
Comment (tonypls) [reply web/src/utils/constants.ts:4]: This was just to test if you're paying attention 😆 Just joking, was testing to see how it looked. I think you're right to keep consistent with the pipes of the commercial website, I think the Map change could be nice or even just `Germany | Electricity Maps` would be nice.

[the remark this answers is not in the document]
Comment (VIKTORVAV99) [reply web/src/utils/constants.ts:4]: I see everything! 👀

File: web/src/features/panels/zone/ZoneHeaderTitle.tsx [hunk 2]
```diff
@@ -31,15 +48,39 @@ export default function ZoneHeaderTitle({ zoneId }: ZoneHeaderTitleProps) {
...
+  const isHourly = useAtomValue(isHourlyAtom);
+  const isMobile = useIsMobile();
 
+  const copyUrlToClipboard = async () => {
+    const url = generateCurrentSelectedDatetimeUrl({ selectedDatetimeString, zoneId });
+    trackShare(ShareType.ZONE);
+
+    if (isMobile && navigator.canShare({ url })) {
+      try {
+        await navigator.share({ title: 'Electricity Maps', text: 'TODO', url: url });
```
Comment (VIKTORVAV99) [inline web/src/features/panels/zone/ZoneHeaderTitle.tsx:62]: Should the title here match the page/meta title we use?

**Comment by tonypls** [comment]
Closing in favour of #7397

**Pull request closed without merging**